exp/views/study.py

```diff
@@ -905,6 +905,18 @@ def dispatch(self, request, *args, **kwargs):
             return HttpResponseRedirect(get_external_url(study, response))
         else:
             _, _, _, study_uuid, _, _, _, *rest = request.path.split("/")
+            """If locale (language code) is present in the URL, then there will be an extra element
+            returned from split("/") before the study UUID, so the UUID variable will be "studies".
+            In this case we need to add an extra returned string before study_uuid to capture the values correctly,
+            and then re-write the request path so that it points to a working study URL.
+            """
+            if study_uuid == "studies":
```

Comment:
I know this isn't the code you originally wrote, but `study_uuid` is confusion.  Could we rename this value?

Reply:
Absolutely.

Reply:
Ah, so looking at this again - I think the `study_uuid` variable name makes sense here, because when there's no locale in the URL, this variable _is_ the study UUID:
```python
_, _, _, study_uuid, _, _, _, *rest = request.path.split("/")
```
But if there is a locale in the URL, then splitting the path returns an extra value that shifts the value-index mapping by 1. Which is why I check here to see if `study_uuid` is actually "studies".

Maybe instead of splitting the path on "/" here, we should use regex to get the locale (if it exists) and study UUID. Thoughts?

Reply:
@okaycj I switched from this `split` method to regex, so there's no longer a variable called `study_uuid` that might _not_ be the study UUID. So I'll mark this one as resolved.